src/main/java/io/stargate/sgv2/jsonapi/service/resolver/matcher/FilterMatcher.java

```diff
@@ -149,7 +171,7 @@ public FilterMatcher<T> compareValues(
     }
   }
 
-  public static final class MatchStrategyCounter {
+  public final class MatchStrategyCounter {
```

Comment:
there is still work to do on this class once we have merged: 

* users of the class need to know if they should strict or greedy, that should be a polymorphic function 
* user of the class needs to know if it should remove from the iterator 

https://github.com/stargate/data-api/issues/1412